addons/hr_holidays/static/src/owl/components/thread_icon/thread_icon.xml

```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<templates xml:space="preserve">
+    <t t-inherit="mail.component.ThreadIcon" t-inherit-mode="extension">
+        <xpath expr="//t[@name='no_im_status']" position="before">
```

Comment:
I don't find it clean to extend from `no_im_status` conditional.
basically we use `t-if`/`t-elif`/`t-else` to ensure unique icon, and there's no really support for a "pattern-matching"-like directive, e.g.`t-case`.
I think it's best not using `t-elif` and add conditions on top instead, so the hooked name is on container of `im_status` icons

Reply:
Replacing `t-if` by `t-elif` would be better indeed if there is no `t-else`, but here the `t-if` that has to replace the `t-else` would have a stupid condition (the inverse of all the others combined, including inherited templates that have to add their own new conditions into the fake "else" too).

The current diff is the best compromise I think.